util/execution/req.go

```diff
@@ -0,0 +1,22 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package execution
+
+import "github.com/pingcap/tidb/util/chunk"
+
+// ExecRequest is input parameter of Executor.Next` method.
+type ExecRequest struct {
```

Comment:
how about change the struct name to "RecordBatch" and make `Chunk` as it's embedded struct:
```go
type RecordBatch struct {
	*chunk.Chunk
}
```

In the outside, we can just call the function bound to `Chunk` directly though an object of `RecordBatch` like this:

```go
	var batch RecordBatch;
	batch.NumRows()
	batch.Reset()
	...
```

Reply:
emm, I have take care about this too, but lost info `Ret`Chunk but ok I can change .